### PR TITLE
113 main window

### PR DESCRIPTION
**Pull request by MaksGMaks**

### CODE REVIEW

File: _client/src/Ui/MainWindow.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <QMainWindow>
+#include <QStackedWidget>
+#include <QStatusBar>
+#include "Ui/WidgetStyles.hpp"
```
Comment (Sage-Cat) [inline _client/src/Ui/MainWindow.hpp]: separate std, Qt includes and ours
Like this:
image

File: _client/src/Ui/WidgetStyles.hpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace SpaceSize
+{
+    const int ADDIT_SPACE_WIDTH = 100;
+    const int ADDIT_SPACE_HEIGHT = 50;
+    const int ADDIT_UND_TAB_SPACE_HEIGHT = 10;
+    const int ADDIT_UND_STAT_SPACE_HEIGHT = 20;
+} // namespace SpaceSize
+
+namespace MainWidSize
```
Comment (Sage-Cat) [inline _client/src/Ui/WidgetStyles.hpp]: MainWid?

Maybe `MainWindowSize`?

File: _client/src/Ui/WidgetStyles.hpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+namespace SpaceSize
+{
+    const int ADDIT_SPACE_WIDTH = 100;
```
Comment (Sage-Cat) [inline _client/src/Ui/WidgetStyles.hpp]: Not clear enough... What is ADDIT?

File: _client/src/UiManager.cpp [hunk 4]
```diff
@@ -72,17 +73,8 @@ void UiManager::initMainWindow()
...
-        m_engine.load(QUrl(QStringLiteral("qrc:/Main.qml")));
-        if (m_engine.rootObjects().isEmpty())
-            SPDLOG_CRITICAL("UiManager::initMainWindow Failed to load main window");
-    }
-    else
-    {
-        SPDLOG_CRITICAL("UiManager::initMainWindow m_engine is invalid");
-    }
+    m_mainWindow = new MainWindow();
+    m_mainWindow->showMaximized();
```
Comment (Sage-Cat) [inline _client/src/UiManager.cpp]: Method exists only for initialization purposes

Do show somewhere else. I recommend to do that in constructor

File: _client/tests/unit/testMainWindow.cpp [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#include <gtest/gtest.h>
+
+#include <QApplication>
+
+#define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_TRACE
```
Comment (Sage-Cat) [inline _client/tests/unit/testMainWindow.cpp]: It's not needed, cos we do that in `SpdlogWrapper`

File: _client/src/Ui/MainWindow.cpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+MainWindow::MainWindow(QWidget *parent)
+    : QMainWindow(parent)
+{
+    init();
+}
+
+void MainWindow::init()
+{
+    // set size
+    this->resize(MainWidSize::WINDOW_WIDTH, MainWidSize::WINDOW_HEIGHT);
```
Comment (Sage-Cat) [inline _client/src/Ui/MainWindow.cpp]: init method exist only for initialization purposes, do this stuff in different place

I recommend to create different method `setupUi`

**Review by Sage-Cat — CHANGES_REQUESTED** [review]

File: _client/src/Ui/MainWindow.hpp (hunk 1 above)
Comment (MaksGMaks) [reply _client/src/Ui/MainWindow.hpp]: done

File: _client/src/Ui/WidgetStyles.hpp (hunk 2 above)
Comment (MaksGMaks) [reply _client/src/Ui/WidgetStyles.hpp]: changed to MainWindowSize

File: _client/src/Ui/WidgetStyles.hpp (hunk 3 above)
Comment (MaksGMaks) [reply _client/src/Ui/WidgetStyles.hpp]: added more information (ADDITION)

File: _client/src/UiManager.cpp (hunk 4 above)
Comment (MaksGMaks) [reply _client/src/UiManager.cpp]: solved

File: _client/tests/unit/testMainWindow.cpp (hunk 5 above)
Comment (MaksGMaks) [reply _client/tests/unit/testMainWindow.cpp]: deleted

File: _client/src/Ui/MainWindow.cpp (hunk 6 above)
Comment (MaksGMaks) [reply _client/src/Ui/MainWindow.cpp]: created setupUI

**Review by Sage-Cat — APPROVED** [review]

**Pull request merged**